libs/stats/odc/stats/tasks.py

```diff
@@ -459,9 +515,29 @@ def stream_from_sqs(
             sqs_queue = get_queue(sqs_queue)
 
         for msg in get_messages(sqs_queue, visibility_timeout=visibility_timeout, **kw):
-            # TODO: switch to JSON for SQS message body
             token = SQSWorkToken(msg, visibility_timeout)
-            tidx = parse_task(msg.body)
+            tidx, filedb = parse_sqs(msg.body)
+
+            # avoid the download and update again, how ever, we have to setup an exception to handle the unit test filedb
+            if urlparse(filedb).scheme == 's3':
+                bucket, key = s3_url_parse(filedb)
+            else: # if it is the test_tiles.db, we load it from local
+                key = filedb 
+            local_cache_file = key.split("/")[-1]
+            if not os.path.isfile(local_cache_file):  # use the download filedb from S3 as the init context flag
+                self.init_from_sqs(filedb)
```

Comment:
does this read the cache file for each message?

Reply:
The control here is: useing cache db_file as 'flag'. 

- if not in local filesystem, we download it from S3, and read it.
- if in local filesystem, we assume the current worker already read it (then skip cache file loading).

Of course, someone may argue 
> if download the s3 file, but application crash. Then the 'cache db_file' flag does not work.

However, consider we are using the K8s pod/docker containter to run it. The application crash will also clean up the application file system. So use local db_file as flag could be safe.